### DISCUSSION TITLE
How to update ui.select options?

### DISCUSSION THREAD

**Discussion by petteri111** [zauberzeug/nicegui#32] [Q&A] [answered]
Hello,

How do you update ui.select options and update the view to user?
Two ui.selects (select1(list1), select2(list2))

in justpy it was async to:

```python
select1.on('input', updatelist)

async def updatelist():
    select2.options =(list2)
    jp.run_task(wp.update())
    await asyncio.sleep(1)
```

How do you make it in nicegui?

**Comment by petteri111** [comment]
I can clear the select2 and write it again (with new options) with callback function, but is there another way?

**Reply by rodja** [reply]
You can access the underlying JustPy object through the `view` property:

```python
def update():
    select.view.options = [5, 6, 7]


select = ui.select([1, 2, 3])
ui.button('update', on_click=update)
```

**Comment by falkoschindler** [comment]
Since NiceGUI 0.8.0 @rodja's answer requires an explicit UI update:

```
def update():
    select.view.options = [5, 6, 7]
    select.update()

select = ui.select([1, 2, 3])
ui.button('update', on_click=update)
```

**Comment by falkoschindler** [comment, marked as answer]
Aforementioned are not recommended anymore because `view.options` no longer represent the `options` the user passed to `ui.select`.

Instead, choice elements like `ui.select`, `ui.toggle` and `ui.radio` now have `options` that can be modified. Calling `element.update()` will update the internal data model and synchronize the new options to the client.
